app/controllers/verify/finance_controller.rb

```diff
@@ -9,7 +9,10 @@ def new
     end
 
     def create
-      if step.complete
+      result = step.submit
```

Comment:
Also, I'm thinking of removing the `Step` name altogether when I'm done refactoring all the steps. I'm thinking something like `ProcessFinanceForm` and have the method be `call`. I think you should be able to tell what a class does from its name only. For example, if I look at the `app/services` folder, I should have a general sense of what the app does. `FinancialsStep` by itself doesn't tell me much, but `ProcessFinanceForm` tells me something is happening after the user submits a form related to finances.